code/game/objects/items/devices/camera_bug.dm

```diff
@@ -7,59 +7,86 @@
 	item_state = ""
 	throw_speed = 4
 	throw_range = 20
-	flags = FPRINT  | NO_ATTACK_MSG
+	flags = FPRINT | NO_ATTACK_MSG
 	var/c_tag = ""
-	var/active = 0
+	var/active = FALSE
 	var/network = ""
-	var/list/excludes = list(/turf/simulated/floor, /turf/space, /turf/simulated/shuttle, /mob/living/carbon, /obj/item/weapon/storage)
-/obj/item/device/camera_bug/attack_self(mob/user)
-	var/newtag = sanitize(input("Set camera tag") as null|text)
+	var/list/excludes = list(
+		/obj/effect,
+		/turf/simulated/floor,
+		/turf/space,
+		/turf/simulated/shuttle,
+		/mob/living/carbon,
+		/obj/item/weapon/storage
+	)
+
+/obj/item/device/camera_bug/attack_self(var/mob/user)
+	var/newtag = sanitize(input(user, "Choose a unique ID tag:", name, c_tag) as null|text)
 	if(newtag)
 		c_tag = newtag
 		if(user.mind)
 			network = "\ref[user.mind]"
 
-/obj/item/device/camera_bug/afterattack(atom/A, mob/user)
+/obj/item/device/camera_bug/afterattack(var/atom/A, var/mob/user, var/proximity_flag)
+	if(!proximity_flag)
+		to_chat(user, "<span class='warning'>You're too far away.</span>")
```

Comment:
"You can't reach it" might be better wording, since proximity is controlled by adjacency too.

Reply:
I'll change this